integration_tests/src/main/python/cache_test.py

```diff
@@ -338,4 +338,9 @@ def do_it(spark):
         df1 = unary_op_df(spark, data_gen).orderBy('a').cache()
         df2 = df1.alias('df2')
         return df1.join(df2, df1.a == df2.a, 'Outer')
-    assert_gpu_and_cpu_are_equal_collect(do_it, conf=conf)
\ No newline at end of file
+    assert_gpu_and_cpu_are_equal_collect(do_it, conf=conf)
+@ignore_order
+@allow_non_gpu("InMemoryTableScanExec", "ProjectExec")
+def test_cache_count():
```

Comment:
Could you add some comments about what this test is trying to do?  Just so that when we do add in timezone support in the future we can update this test appropriately.

Perhaps you could split it up into two queries. The first one runs on the CPU only and caches the data, and the second one runs on the GPU to do the count.  That way we don't have to worry about time zone support changing, but we still want the comments.

Reply:
If we support timezones in the future this test case will not happen. The reason why this is happening is because InMemoryTableScanExec is falling back to the CPU. With timezones supported the InMemoryTableScanExec will not fallback to the CPU upon cache read and return data already on the GPU which won't cause this problem to happen. It won't matter if we create the cache on CPU or not. At least that's how I understand it. 

I will add the comments for now until you come back and give